### PR TITLE
Fixing errors in nonstandardcode.py using vscode

### LINKED ISSUE #5: Fixing errors and merge conflicts-Assignment1.3
- Create a GitHub issue for the current assignment and add in the details based on what is requested in this assignment and assign it to yourself.

- Install VSCode and configure it for for python development

- Install the following plugins:
      1. Python
      2. Remote-WSL
      3. Git lens

- Configure VSCode for Python development

- Ensure the following tools are configured in your user settings.
    1. Linter: use flake8
    2. Sorted Imports: use isort
    3. Auto-formatter: use black
    4. Python code defaults: 4 spaces for tab, remove trailing whitespaces
  

- Ensure the following configurations are set in your workspace settings
    1. Python Path
    2. Run auto-formatter and sort imports everytime the file is saved.
    3. Custom parameters to the code-formatting tools (if needed)

- Create a PR with the auto-formatted code.

- Create a new branch off of the git tag `v0.1` named `fix/<issue-id>/vscode-refactor`.

- Fix the problems highlighted in VSCode until it indicates there are no problems with the script.

- Create a PR from the new branch

- Fix any merge conflicts

### PR DESCRIPTION
**Pull request by Dhikshita-macherla**
Hence closes #5 
-Created an issue
-installed all the necessary packages in vs code
-Configured VSCode for Python development
- Configured the user and workspace settings
- Created a PR for the changes

Commands
$git checkout -b fix/5/vscode-refactor v0.1
$nano nonstandardcode.py
$ flake8 nonstandardcode.py
$black nonstandardcode.py
$isort nonstandardcode.py
$git add nonstandardcode.py
$ git commit -am "Fixed errors in nonstandardcode.py"
$git push origin fix/5/vscode-refactor

Screenshot
image

Workspace settings
{
    "python.defaultInterpreterPath": "/home/dhikshita/miniconda3/bin/python",
    "editor.formatOnSave": false,
    "[python]": {
        "editor.formatOnType": true,
        "editor.codeActionsOnSave": {
            "source.organizeImports": "explicit"
        },
    },
    "flake8.args": [
        "--max-line-length=79"
    ],
    "black-formatter.args": ["--line-length", "79"],
    "isort.args" : ["--profile", "black", "--float-to-top"
    ]
}

User setting
{
    "isort.check": true,
    "python.linting.enabled": true,
    "python.linting.flake8Enabled": true,
    "[python]": {
        "editor.tabSize": 4,
        "editor.defaultFormatter": "ms-python.black-formatter",
    },
    "files.trimTrailingWhitespace": true,
   
}

nonstandardcode.py output
$python nonstandardcode.py
image

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    "editor.formatOnSave": false,
+    "[python]": {
+        "editor.formatOnType": true,
+        "editor.codeActionsOnSave": {
+            "source.organizeImports": "explicit"
+        },
+    },
+    "flake8.args": [
+        "--max-line-length=120"
+    ],
```
Comment (abilashravi-ta) [inline .vscode/settings.json:12]: Why is this `max-line-length` set to **120** while in `black` you have kept it as **79**.

**Review by abilashravi-ta — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 1 above)
Comment (Dhikshita-macherla) [reply .vscode/settings.json:12]: While executing flake8 nonstandardcode.py, it showed there were more than 79 characters in a single line. So, I changed the flake8 arguments

Comment (abilashravi-ta) [reply .vscode/settings.json:12]: Make the `max-line-length` as **79** for both `black` and `flake8`.

**Review by abilashravi-ta — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 1 above)
Comment (Dhikshita-macherla) [reply .vscode/settings.json]: I've changed all the corrections, Can you check if its alright?

**Review by abilashravi-ta — APPROVED** [review]
LGTM.

**Pull request merged**